tests/bsim/bluetooth/host/att/sequential/tester/src/main.c

```diff
@@ -0,0 +1,715 @@
+	__ASSERT_NO_MSG(buf);
+
+	reserve = sizeof(struct bt_l2cap_hdr);
+	reserve += sizeof(struct bt_hci_acl_hdr) + BT_BUF_RESERVE;
+
+	net_buf_reserve(buf, reserve);
+
+	return buf;
+}
+
+static int send_acl(struct net_buf *buf)
+{
+	struct bt_hci_acl_hdr *hdr;
+	uint8_t flags = BT_ACL_START_NO_FLUSH;
+
+	hdr = net_buf_push(buf, sizeof(*hdr));
+	hdr->handle = sys_cpu_to_le16(bt_acl_handle_pack(conn_handle, flags));
+	hdr->len = sys_cpu_to_le16(buf->len - sizeof(*hdr));
+
+	bt_buf_set_type(buf, BT_BUF_ACL_OUT);
+
+	k_sem_take(&acl_pkts, K_FOREVER);
+
+	return bt_send(buf);
+}
+
+static void send_l2cap_packet(struct net_buf *buf, uint16_t cid)
+{
+	struct bt_l2cap_hdr *hdr;
+
+	hdr = net_buf_push(buf, sizeof(*hdr));
+	hdr->len = sys_cpu_to_le16(buf->len - sizeof(*hdr));
+	hdr->cid = sys_cpu_to_le16(cid);
+
+	/* Always entire packets, no HCI fragmentation */
+	__ASSERT_NO_MSG(buf->len <= CONFIG_BT_BUF_ACL_TX_SIZE);
+
+	send_acl(buf);
+}
+
+static void gatt_write(bool req)
+{
+	static uint8_t data[] = "write";
+	uint16_t handle = server_write_handle;
+	struct net_buf *buf = alloc_l2cap_pdu();
+
+	net_buf_add_u8(buf, req ?
+		       BT_ATT_OP_WRITE_REQ : BT_ATT_OP_WRITE_CMD);
+
+	net_buf_add_le16(buf, handle);
+	net_buf_add_mem(buf, data, sizeof(data));
+
+	LOG_INF("send ATT write %s", req ? "REQ" : "CMD");
+	send_l2cap_packet(buf, BT_L2CAP_CID_ATT);
+}
+
+static void gatt_notify(void)
+{
+	static uint8_t data[] = "notification";
+	uint16_t handle = HVX_HANDLE;
+	struct net_buf *buf = alloc_l2cap_pdu();
+
+	net_buf_add_u8(buf, BT_ATT_OP_NOTIFY);
+	net_buf_add_le16(buf, handle);
+	net_buf_add_mem(buf, data, sizeof(data));
+
+	LOG_INF("send ATT notification");
+	send_l2cap_packet(buf, BT_L2CAP_CID_ATT);
+}
+
+static void gatt_indicate(void)
+{
+	static uint8_t data[] = "indication";
+	uint16_t handle = HVX_HANDLE;
+	struct net_buf *buf = alloc_l2cap_pdu();
+
+	net_buf_add_u8(buf, BT_ATT_OP_INDICATE);
+	net_buf_add_le16(buf, handle);
+	net_buf_add_mem(buf, data, sizeof(data));
+
+	LOG_INF("send ATT indication");
+	send_l2cap_packet(buf, BT_L2CAP_CID_ATT);
+}
+
+static void init_ll(void)
+{
+	/* Initialize controller */
+	struct net_buf *rsp;
+
+	send_cmd(BT_HCI_OP_RESET, NULL, NULL);
+	send_cmd(BT_HCI_OP_LE_READ_BUFFER_SIZE, NULL, &rsp);
+	le_read_buffer_size_complete(rsp);
+
+	set_data_len();
+	set_event_mask(BT_HCI_OP_SET_EVENT_MASK);
+	set_event_mask(BT_HCI_OP_LE_SET_EVENT_MASK);
+	set_random_address();
+}
+
+static void init_tinyhost(void)
+{
+	bt_enable_raw(&rx_queue);
+
+	/* Start the RX thread */
+	k_thread_create(&rx_thread_data, rx_thread_stack,
+			K_THREAD_STACK_SIZEOF(rx_thread_stack), rx_thread,
+			NULL, NULL, NULL, K_PRIO_PREEMPT(0), 0, K_NO_WAIT);
+	k_thread_name_set(&rx_thread_data, "HCI RX");
+
+	k_thread_priority_set(k_current_get(), K_PRIO_PREEMPT(0));
+
+	init_ll();
+}
+
+void test_procedure_0(void)
+{
+	init_tinyhost();
+
+	/* Start advertising & wait for a connection */
+	start_adv(60); /* Interval doesn't matter */
```

Comment:
If the interval doesn't matter, then you might as well omit it as a parameter and hardcode it in the `start_adv` function instead

Reply:
yup, again this file has been re-used that's why.